code/game/objects/structures.dm

```diff
@@ -27,6 +30,20 @@
 		queue_smooth_neighbors(src)
 	return ..()
 
+/obj/structure/deconstruct(disassembled = TRUE)
+	if(!disassembled)
+		if(!(flags_1 & NODECONSTRUCT_1))
+			if(islist(debris))
+				for(var/I in debris)
+					for(var/i in 1 to debris[I])
+						new I (get_turf(src))
+		if(break_message)
+			visible_message(break_message)
+		if(break_sound)
+			playsound(src, break_sound, 50, 1)
+	qdel(src)
+	return 1
```

Comment:
```suggestion
	return TRUE
```

Reply:
i didnt think of fixing the original badcode